src/app/_services/api/auth.api.ts

```diff
@@ -45,4 +45,26 @@ export const logout = ({ email, token }: AuthRequest): Promise<void> => (
 
     return res.json();
   })
-)
\ No newline at end of file
+);
+
+export const join = (form: JoinForm): Promise<void> => (
+  fetch(`${baseURL}${AUTH}`, {
+    body: JSON.stringify(form),
+  }).then((res) => {
+    if (!res.ok) {
+      return res.json()
+        .then(Promise.reject.bind(Promise))
```

Comment:
.then((response) => Promise.reject(response));

.then(Promise.reject)

.map(Number)

const reject() => {
  this=> window
}